IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -780,51 +769,60 @@ - (MSIDBaseToken *)getRefreshTokenByLegacyUserId:(NSString *)legacyUserId
+                                                                                              legacyUserId:legacyUserId
+                                                                                                   context:context
+                                                                                                     error:error];
 
-        if ([matchedIdTokens count])
-        {
-            NSString *homeAccountId = matchedIdTokens[0].homeAccountId;
+    if ([matchedIdTokens count])
+    {
+        MSIDCredentialCacheItem *matchedIdToken = matchedIdTokens[0];
+        NSString *homeAccountId = matchedIdToken.homeAccountId;
 
-            MSIDDefaultCredentialCacheQuery *rtQuery = [MSIDDefaultCredentialCacheQuery new];
-            rtQuery.homeAccountId = homeAccountId;
-            rtQuery.environment = alias.msidHostWithPortIfNecessary;
-            rtQuery.clientId = clientId;
-            rtQuery.familyId = familyId;
-            rtQuery.credentialType = MSIDRefreshTokenType;
+        MSID_LOG_VERBOSE(context, @"(Default accessor] Found Match with environment %@, realm %@, client %@", matchedIdToken.environment, matchedIdToken.realm, matchedIdToken.clientId);
+        MSID_LOG_VERBOSE_PII(context, @"(Default accessor] Found Match with environment %@, realm %@, client %@, home account ID %@", matchedIdToken.environment, matchedIdToken.realm, matchedIdToken.clientId, matchedIdToken.homeAccountId);
 
-            NSArray<MSIDCredentialCacheItem *> *rtCacheItems = [_accountCredentialCache getCredentialsWithQuery:rtQuery
-                                                                                              legacyUserId:nil
-                                                                                                   context:context
-                                                                                                     error:error];
+        MSIDDefaultCredentialCacheQuery *rtQuery = [MSIDDefaultCredentialCacheQuery new];
+        rtQuery.homeAccountId = homeAccountId;
+        rtQuery.environmentAliases = aliases;
+        rtQuery.clientId = clientIdForQueries;
+        rtQuery.familyId = familyId;
+        rtQuery.credentialType = MSIDRefreshTokenType;
 
-            if ([rtCacheItems count])
-            {
-                MSID_LOG_VERBOSE(context, @"(Default accessor) Found %lu refresh tokens", (unsigned long)[rtCacheItems count]);
-                MSIDCredentialCacheItem *resultItem = rtCacheItems[0];
-                MSIDBaseToken *resultToken = [resultItem tokenWithType:MSIDRefreshTokenType];
-                resultToken.storageAuthority = resultToken.authority;
-                resultToken.authority = authority;
-                [MSIDTelemetry stopCacheEvent:event withItem:resultToken success:YES context:context];
-                return resultToken;
-            }
+        NSArray<MSIDCredentialCacheItem *> *rtCacheItems = [_accountCredentialCache getCredentialsWithQuery:rtQuery
```

Comment:
I don't see how this one is changed, but I am guessing it will check for all aliases within the query method, correct?